app/api/server/v1/roles.js

```diff
@@ -44,12 +46,29 @@ API.v1.addRoute('roles.create', { authRequired: true }, {
 			description: this.bodyParams.description,
 		};
 
-		Meteor.runAsUser(this.userId, () => {
-			Meteor.call('authorization:saveRole', roleData);
-		});
+		if (!hasPermission(Meteor.userId(), 'access-permissions')) {
+			throw new Meteor.Error('error-action-not-allowed', 'Accessing permissions is not allowed');
+		}
+
+		if (Roles.findOneByIdOrName(roleData.name)) {
+			throw new Meteor.Error('error-duplicate-role-names-not-allowed', 'Role name already exists');
+		}
+
+		if (['Users', 'Subscriptions'].includes(roleData.scope) === false) {
+			roleData.scope = 'Users';
+		}
+
+		const roleId = Roles.createWithRandomId(roleData.name, roleData.scope, roleData.description, false, roleData.mandatory2fa);
```

Comment:
`roleData.mandatory2fa` needs to be declared on the check's schema for it to be a valid param (first line of the method).